core/string/ustring.cpp

```diff
@@ -5365,64 +5409,16 @@ bool String::is_relative_path() const {
 }
 
 String String::get_base_dir() const {
-	int end = 0;
+	const int drive_end_idx = find_drive_end_idx(*this);
+	const int last_separator_idx = MAX(rfind_char('/'), rfind_char('\\'));
```

Comment:
I would raise a point, as we are in an optimize PR. you can do this with less operations again( but more lines). Here you are doing two `rfind`, but you can do only one reverse loop yourself to find this. No function calls, no max.

Reply:
I agree, but before I optimize every situation like that by hand, I'd rather implement `rfind_chars` to not repeat logic, which is definitely outside the scope of this PR.

Reply:
I don't know if you'll ever have exactly this case as a function though, bacause you need a reverse find for multiple characters and being able to specify the minimum index. Seems far fetched.

Reply:
The function would be `rfind(StrRange str, StrRange chars)` which i don't think is far fetched at all :)

Reply:
yeah you are right that would be better. So let's keep that in mind for later. You can resolve this !